### PR TITLE
Sprint 49

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 49

[6069: Apply Read/Unread to New Messaging Flow, and total counts to QC boxes](https://github.com/flexion/ef-cms/issues/6069)
[795: Petitioner spouse service indicator should update to "None" when they are represented by private Practitioner](https://trello.com/c/i6WTABii/795-petitioner-spouse-service-indicator-should-update-to-none-when-they-are-represented-by-private-practitioner)
[802: Page Count on Docket Record incorrect for Notice of Change of Address](https://trello.com/c/XwCjvAVg/802-page-count-on-docket-record-incorrect-for-notice-of-change-of-address)
[815: Upload PDF - on save not selecting the uploaded PDF on drafts list](https://trello.com/c/n7JuxJHi/815-upload-pdf-on-save-not-selecting-the-uploaded-pdf-on-drafts-list)
[807: Practitioner emails not displaying](https://trello.com/c/vkJFr1mR/807-practitioner-emails-not-displaying)
[809: Duplicate language on Notice of Docket Change for external parties](https://github.com/flexion/ef-cms/issues/6442)
[6444: Case that was scheduled for trial session but not checked off as QC'd is still showing as Scheduled after calendar is set](https://github.com/flexion/ef-cms/issues/6444)
[6452: PrivatePractitioner needs a firmName](https://github.com/flexion/ef-cms/issues/6452)
[6466: Drafts screen - There should never be no document selected](https://github.com/flexion/ef-cms/issues/6466)
[6438: Completed work items not displaying in Processed tab](https://github.com/flexion/ef-cms/issues/6438)
[6440: Notice of Docket Change / Notice of Change of Address/ Notice of Trial/ Standing Pre Trial Order or Notice - these should be hyperlinked on Docket Record after it's added and page count is wrong](https://github.com/flexion/ef-cms/issues/6440)
[6465: Upload PDF - when changing PDF, the new PDF is not displaying](https://github.com/flexion/ef-cms/issues/6465)

### CODE REVIEW

File: shared/src/business/entities/EntityValidationConstants.js [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    ...CHAMBERS_SECTIONS,
+    ...Object.values(ROLES),
+  ).optional(),
+  sentByUserId: JoiValidationConstants.UUID.optional(),
+  trialDate: JoiValidationConstants.ISO_DATE.optional().allow(null),
+  updatedAt: JoiValidationConstants.ISO_DATE.required(),
+  workItemId: JoiValidationConstants.UUID.required(),
+};
+
+// TODO: validate workItems in DocketEntry
```
Comment (julialeague) [inline shared/src/business/entities/EntityValidationConstants.js:415]: Is there a story in the backlog for this, and could it be referenced here?

Comment (julialeague) [reply shared/src/business/entities/EntityValidationConstants.js:415]: Never mind, I see it at the top of the file now.

File: shared/src/business/entities/WorkItem.test.js [hunk 2]
```diff
@@ -115,4 +115,71 @@ describe('WorkItem', () => {
...
+        docketNumber: '101-18',
+        docketNumberSuffix: DOCKET_NUMBER_SUFFIXES.SMALL,
+        section: DOCKET_SECTION,
+        sentBy: 'bob',
+      },
+      { applicationContext },
+    );
+    expect(workItem.highPriority).toBe(false);
+
+    workItem = new WorkItem(
```
Comment (julialeague) [inline shared/src/business/entities/WorkItem.test.js:136]: This expect block appears to occur twice - I think that's a mistake and we could remove one, right?

File: shared/src/business/useCases/sealCaseContactAddressInteractor.test.js [hunk 3]
```diff
@@ -39,6 +39,30 @@ describe('sealCaseContactAddressInteractor', () => {
...
+      role: ROLES.docketClerk,
+      userId: 'docketClerk',
+    });
+    await expect(
+      sealCaseContactAddressInteractor({
+        applicationContext,
+        contactId: '23-skidoo',
+        docketNumber: MOCK_CASE.docketNumber,
+      }),
+    ).rejects.toThrow('Cannot seal contact');
```
Comment (julialeague) [inline shared/src/business/useCases/sealCaseContactAddressInteractor.test.js:63]: I'm not understanding the connection between the description of this test and the expectation for an error being thrown here - is an error thrown because of the empty arrays for `otherFilers`/`otherPetitioners`?

Comment (ericsorenson) [reply shared/src/business/useCases/sealCaseContactAddressInteractor.test.js:63]: Updated!

File: shared/src/business/entities/WorkItem.test.js (hunk 2 above)
Comment (ericsorenson) [reply shared/src/business/entities/WorkItem.test.js:136]: Updated!

**Comment by julialeague** [comment]
- [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
- [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
		- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
	- [x] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Review by julialeague — APPROVED** [review]
@mmarcotte I recommend you accept this PR into the USTC environments. Thanks!
